app/components/at_mention/index.js

```diff
@@ -3,18 +3,23 @@
 
 import {connect} from 'react-redux';
 
-import {getUsersByUsername, getCurrentUserMentionKeys} from '@mm-redux/selectors/entities/users';
+import {getUsersByUsername} from '@mm-redux/selectors/entities/users';
+
+import {getAllUserMentionKeys} from '@mm-redux/selectors/entities/search';
 
 import {getTeammateNameDisplaySetting, getTheme} from '@mm-redux/selectors/entities/preferences';
 
+import {getGroupsByName} from '@mm-redux/selectors/entities/groups';
+
 import AtMention from './at_mention';
 
-function mapStateToProps(state) {
+function mapStateToProps(state, ownProps) {
     return {
         theme: getTheme(state),
         usersByUsername: getUsersByUsername(state),
-        mentionKeys: getCurrentUserMentionKeys(state),
+        mentionKeys: ownProps.mentionKeys || getAllUserMentionKeys(state),
```

Comment:
are you  expecting `ownProps.mentionKeys` to be different than `getAllUserMentionKeys` ?

Reply:
Yes it could be because mentionKeys is retrieved from `makeGetMentionKeysForPost` which could be missing some mentionKeys depending on the boolean values passed. However, as you mentioned, since it's required in this file we can just do a `ownProps.mentionKeys`.

Reply:
Same thing as above. `<AtMention/>` is being called from other places other than <Markdown/> so if I'm not passing the mention Keys, I need to just get all of them.